### ISSUE TITLE
yaml editor not taking correct api Version

### ISSUE THREAD

**Issue by mrSingh007** [lensapp/lens#978]
**Describe the bug**
when the yaml file is created or applied with current api version, then the lens yaml editor not showing that file in correct version. 
for example following ingress is created from bash on machine: 
```
apiVersion: networking.k8s.io/v1
kind: Ingress
```
and if same file is opened in lens yaml editor then it is shown as:-
```
kind: Ingress
apiVersion: extensions/v1beta1  #<--- WRONG
```
So it have changed the apiVersion. Now doing any change to yaml file through lens will show warning that this api
extension/v1beta1 is deprecated 

**To Reproduce**
 create ingress kind file with 'apiVersion: extensions/v1beta1'. Now open lens and  go to network-ingresses. Here from three dots try to edit it. 
- Api version will not be correct. 
- when saving changes warning will be shown

**Expected behavior**
It should take current api version version from yaml file

**Environment (please complete the following information):**
- Lens Version: 3.6.r
- OS: MacOS catalina

**Comment by Nokel81** [comment]
Is this still an issue with either 4.2.4 or 5.0.0-beta.9?

**Comment by Monta89** [comment]
I'm currently using version 5.2.5 and noticing this bug!

**Comment by 9digitdev** [comment]
I'm using version 5.3.4 and I'm still seeing the issue

**Comment by Nokel81** [comment]
This should be fixed by #4406, targeted currently for 5.4

**Comment by Nokel81** [comment]
This is because there is no "saved version" in kubernetes and all entries are converted to the version that they are requested on.

**Comment by zlodes** [comment]
Same with CronJobs.
![image](https://user-images.githubusercontent.com/1114387/214580388-8d4ec81f-6cce-4c0e-ada1-76564ebc341a.png)

**Comment by Nokel81** [comment]
Cron jobs was fixed by #6542 and will be part of the next release

**Comment by Nokel81** [comment]
Closing in favour of https://github.com/lensapp/lens/issues/7474 for specifically that feature. The auto detection of new version should be fixed separately.
